app/jobs/scheduled/update_user_counts_for_chat_channels.rb

```diff
@@ -2,25 +2,25 @@
 
 module Jobs
   class UpdateUserCountsForChatChannels < ::Jobs::Scheduled
-    every 2.hours
+    every 1.hour
 
+    # FIXME: This could become huge as the amount of channels grows, we
+    # need a different approach here. Perhaps we should only bother for
+    # channels updated or with new messages in the past N days? Perhaps
+    # we could update all the counts in a single query as well?
     def execute(args = {})
-      ChatChannel.find_each { |chat_channel| set_user_count(chat_channel) }
+      ChatChannel
+        .where(status: %i[open closed])
+        .find_each { |chat_channel| set_user_count(chat_channel) }
     end
 
     def set_user_count(chat_channel)
       current_count = chat_channel.user_count || 0
-      new_count =
-        chat_channel
-          .user_chat_channel_memberships
-          .joins(:user)
-          .where(following: true)
-          .merge(User.activated.not_suspended.not_staged)
-          .count
-
+      new_count = ChatChannelMembershipsQuery.call(chat_channel, count_only: true)
```

Comment:
Wonder if signature shouldn't be: `ChatChannelMembershipsQuery.count(channel)` ?

Reply:
Yes that's better, will change